databrowser/src/pages/datasets/overview/OverviewListPage.vue

```diff
@@ -366,12 +383,50 @@ const hideFilters = () => {
   isFiltersModalVisible.value = false;
 };
 
+const filterSelectedForComponent = ref<{ key: string; value: string }[]>([]);
+
+const initializeFiltersAndSearch = () => {
+  const {rawfilter, searchQuery} = getStartedQuery();
+
+  if (!searchQuery && !rawfilter) {
+    return;
+  }
+
+  if (searchQuery && searchQuery !== '') {
```

Comment:
Isn't this just a check for string emptiness? Watch out for implicit type casts...